data-analysis/13-data-processing.ipynb

```diff
@@ -0,0 +1,1803 @@
+    "\n",
+    "- A fast and efficient DataFrame object for data manipulation with integrated indexing\n",
+    "\n",
+    "- Tools for reading and writing data between in-memory data structures and different formats: CSV and text files, Microsoft Excel, SQL databases, and the fast HDF5 format\n",
+    "\n",
+    "- Flexible reshaping and pivoting of data sets\n",
+    "\n",
+    "- Columns can be inserted and deleted from data structures for size mutability\n",
+    "\n",
+    "- Aggregating or transforming data with a powerful group by engine allowing split-apply-combine operations on data sets;\n",
+    "\n",
+    "- High performance merging and joining of data sets;\n",
+    "\n",
+    "- Hierarchical axis indexing provides an intuitive way of working with high-dimensional data in a lower-dimensional data structure;    "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "wanted-blink",
+   "metadata": {},
+   "source": [
+    "# If Pandas is so great why do we need anything else? \n",
+    "The popularity of Pandas is largely due to it's maturity and time in use but, it was written to use a single processor and loads the entire dataframe into memory thus, severely lowering it's scalability and efficiency. In the last few years the open source community has developed modules built on top of Pandas like Dask, Ray, and Modin which run python code across compute clusters or parallelize code to run on several clusters. RAPIDs for example, is a partial Pandas replacement that runs code on GPUs. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "conceptual-edinburgh",
+   "metadata": {},
+   "source": [
+    "![ModinVDataframe](../img/mostcommonpandas.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "blind-october",
+   "metadata": {},
+   "source": [
+    "## Introduction to Modin\n",
+    "Modin was designed for terrabyte sized data sets by splitting the dataframe into smaller more manageble units allowing it to be loaded into separate processors and memory thus, parallelizing your Python code. \n",
+    "It covers 90% of the Pandas API with a focus on the most commonly used Pandas methods. Modin is a superset of Pandas which means the remaining falls back to native Pandas. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "blond-violin",
+   "metadata": {},
+   "source": [
+    "![ModinVDataframe](../img/PandasVModin1.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "color-berkeley",
+   "metadata": {},
+   "source": [
+    "# Using Modin\n",
+    "Modin includes a scheduler that abstracts the hardware from the interpreter so that you don't need to know manage it within your code.\n",
+    "To install it all you need: <br />\n",
+    "    `pip install modin[ray]`<br />\n",
+    "    `pip install modin[dask]`<br />\n",
+    "    `pip install modin[all]`\n",
```

Comment:
we tend not to put a pin on how a certain 3rd party library is installed, instead, we usually link to the official website since these things can change in the future

Reply:
@mannympardo

Reply:
Almost every article I read referenced the code but, I'm fine removing it.

Reply:
reference to the link, since these things can change